libethereum/SnapshotImporter.cpp

```diff
@@ -0,0 +1,257 @@
+/*
+	This file is part of cpp-ethereum.
+
+	cpp-ethereum is free software: you can redistribute it and/or modify
+	it under the terms of the GNU General Public License as published by
+	the Free Software Foundation, either version 3 of the License, or
+	(at your option) any later version.
+
+	cpp-ethereum is distributed in the hope that it will be useful,
+	but WITHOUT ANY WARRANTY; without even the implied warranty of
+	MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+	GNU General Public License for more details.
+
+	You should have received a copy of the GNU General Public License
+	along with cpp-ethereum.  If not, see <http://www.gnu.org/licenses/>.
+*/
+
+#include "SnapshotImporter.h"
+#include "Client.h"
+#include "SnapshotStorage.h"
+
+#include <libdevcore/RLP.h>
+#include <libdevcore/TrieHash.h>
+#include <libethashseal/Ethash.h>
+
+#include <snappy.h>
+
+using namespace dev;
+using namespace eth;
+
+namespace
+{
+
+struct SnapshotImportLog: public LogChannel
+{
+	static char const* name() { return "SNAP"; }
+	static int const verbosity = 9;
```

Comment:
Were you able to use it for debugging? I find our logging system without filters quite useless.

Reply:
Yeah. On Windows it somehow looks more clear, at least in release build and I can see snapshot importing progress. On Linux there's more spam from somewhere (probably `BlockChain`)